### DISCUSSION TITLE
mdbook serve fails on cygwin

### DISCUSSION THREAD

**Discussion by kieranshanley** [google/comprehensive-rust#681] [Q&A] [answered]
Got as far as running up mdbook serve and got the following error:

$ mdbook serve
2023-05-25 17:02:08 [INFO] (mdbook::book): Book building has started
2023-05-25 17:02:08 [INFO] (mdbook::book): Running the exerciser backend
2023-05-25 17:02:08 [INFO] (mdbook::renderer): Invoking the "exerciser" renderer
2023-05-25 17:02:08 [INFO] (mdbook::book): Running the html backend
2023-05-25 17:02:10 [ERROR] (mdbook::utils): Error: Rendering failed
2023-05-25 17:02:10 [ERROR] (mdbook::utils):    Caused By: Failed to read "C:\\cygwin64\\home\\shanleyk1\\code\\comprehensive-rust\\src\\android\\interoperability\\cpp\\overview.svg"

It looks like an issue with following links as I can see the file is there and I was able to open it:

 ls -l "C:\\cygwin64\\home\\shanleyk1\\code\\comprehensive-rust\\src\\android\\interoperability\\cpp\\overview.svg"
lrwxrwxrwx 1 LEIDOS-CORP+shanleyk1 LEIDOS-CORP+Domain Users 40 May 25 16:10 'C:\cygwin64\home\shanleyk1\code\comprehensive-rust\src\android\interoperability\cpp\overview.svg' -> ../../../../third_party/cxx/overview.svg

Is there a trick to getting the symbolic links to work under cygwin ?

cheers,

Kieran

**Comment by mgeisler** [comment, marked as answer]
Hi Kieran,

I'm afraid I've never tried serving the course under Windows (in any form, Cygwin or not). So I had not considered that the symlinks could cause problems for you.

What you could do would be to copy the `overview.svg` file from the `third_party/` folder to the `android/interoperability/cpp/` folder by hand. That should let you build the book locally. Alternatively, you could remove the reference from `cpp.md`. You'll lose the picture [here](https://google.github.io/comprehensive-rust/android/interoperability/cpp.html), but you should otherwise be able to read and change the book.

**Comment by kieranshanley** [comment]
Great I followed your advice and manually copied over the svg and it solved my problem. Thanks for the quick response

**Reply by mgeisler** [reply]
Glad it worked! It's been many years since I was near a Cygwin environment — I'm sure people who work with it has a standard way to deal with symlinks, but I don't know it :slightly_smiling_face:

**Reply by mgeisler** [reply]
The discussion in #733 might be relevant here: Git has some options for how to handle symlinks and perhaps they can help on Cygwin too.
